### PR TITLE
QueueReceiver only allocates once for empty receives and returns a list

### PR DESCRIPTION
**Pull request by danielmarbach**
instead of an enumerable to save allocations

### CODE REVIEW

File: src/Transport/AzureMessageQueueReceiver.cs [hunk 1]
```diff
@@ -54,15 +54,20 @@ public AzureMessageQueueReceiver(MessageEnvelopeUnwrapper unwrapper, CloudQueueC
...
         }
 
-        internal async Task<IEnumerable<MessageRetrieved>> Receive(CancellationToken token)
+        internal async Task<List<MessageRetrieved>> Receive(CancellationToken token)
         {
             var rawMessages = await azureQueue.GetMessagesAsync(BatchSize, MessageInvisibleTime, null, null, token).ConfigureAwait(false);
 
             var messageFound = false;
-            var messages = new List<MessageRetrieved>();
+            var messages = noMessagesFound;
```
Comment (Scooletz) [inline src/Transport/AzureMessageQueueReceiver.cs]: What if we assinged `null` in here:
```
List<MessageRetrieved> messages = null;
```
and return
``` 
returns messages ?? noMessagesFound;
```
IMO this is better as you won't accidentally introduce a change that changes `noMessagesFound`

Comment (danielmarbach) [inline src/Transport/AzureMessageQueueReceiver.cs]: Good for me

Comment (Scooletz) [inline src/Transport/AzureMessageQueueReceiver.cs]: Will you issue an update?

**Comment by danielmarbach** [comment]
@Scooletz pushed

**Pull request merged**